src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -2134,137 +2799,196 @@ public enum RisingRivals implements LogicCardInfo {
+            def tar = my.all.findAll{it != src}.select("Choose Pokémon to move energy cards to")
             energyList.each {
               energySwitch(src, tar, it)
             }
           }
           playRequirement{
-            assert my.all.filterByType(ENERGY) : "You don't have any energy in play"
+            assert my.all.find{it.cards.energyCount(C)} : "You don't have any energy in play"
           }
         };
       case POKEMON_CONTEST_HALL_93:
-        return basicTrainer (this) {
+        return stadium (this) {
           text "This card stays in play when you play it. Discard this card if another Stadium card comes into play. If another card with the same name is in play, you can’t play this card.\nOnce during each player’s turn, if that player’s Bench isn’t full, the player may flip a coin. If heads, that player searches his or her deck for a Basic Pokémon and puts it onto his or her Bench. If the player does, he or she may search his or her deck for a Pokémon Tool card and attach it to that Pokémon. If that player searched his or her deck, the player shuffles his or her deck afterward."
+          def lastTurn=0
+          def actions=[]
           onPlay {
+            actions=action("Stadium: Pokemon Contest Hall") {
+              assert my.deck
+              assert my.bench.notFull
+              assert lastTurn != bg().turnCount : "Already used"
+              bc "Used Pokemon Contest Hall effect"
+              lastTurn = bg().turnCount
+              flip {
+                my.deck.search {it.cardTypes.is(BASIC)}.each {
+                  def pcs = benchPCS(it)
+                  my.deck.search {it.cardTypes.is(POKEMON_TOOL)}.each {
+                    attachPokemonTool(it,pcs)
+                  }
+                }
+                shuffleDeck()
+              }
+            }
           }
-          playRequirement{
+          onRemoveFromPlay{
+            actions.each { bg().gm().unregisterAction(it) }
           }
         };
       case SUNYSHORE_CITY_GYM_94:
-        return basicTrainer (this) {
+        return stadium (this) {
           text "This card stays in play when you play it. Discard this card if another Stadium card comes into play. If another card with the same name is in play, you can’t play this card.\nAny damage done by attacks from [L] Pokémon (both yours and your opponent’s) to the Defending Pokémon isn’t affected by Resistance. Each [L] Pokémon in play (both yours and your opponent’s) has no Weakness."
+          def eff1, eff2
           onPlay {
+            eff1 = getter (GET_WEAKNESSES) {h->
+              if(h.effect.target.types.contains(L)){
+                h.object = []
+              }
+            }
+            eff2 = delayed {
+              before APPLY_RESISTANCE, {
+                bg.dm().each {
+                  if (it.from.types.contains(L) && it.to.active) {
+                    prevent()
+                  }
+                }
+              }
+            }
```

Comment:
Does this work right? I think I'd prefer in instead `after PROCESS_ATTACK_EFFECTS` if you added the NO_RESISTANCE flag instead (see Phoebe from BattleStyles for reference adding a flag)

Reply:
This does work, but I agree using a NO_RESISTANCE flag is better here